shared_model/validators/commands_validator.hpp

```diff
@@ -0,0 +1,303 @@
+          std::string class_name = "GrantPermission";
+          ReasonsGroupType res;
+
+          validateAccountId(res, gp->accountId());
+
+          return res;
+        }
+
+        ReasonsGroupType operator()(
+            const detail::PolymorphicWrapper<interface::RemoveSignatory> &rs)
+            const {
+          std::string class_name = "RemoveSignatory";
+          ReasonsGroupType res;
+
+          validateAccountId(res, rs->accountId());
+          validatePubkey(res, rs->pubkey());
+
+          return res;
+        }
+        ReasonsGroupType operator()(
+            const detail::PolymorphicWrapper<interface::RevokePermission> &rp)
+            const {
+          std::string class_name = "RevokePermission";
+          ReasonsGroupType res;
+
+          validateAccountId(res, rp->accountId());
+          validatePermission(res, rp->permissionName());
+
+          return res;
+        }
+        ReasonsGroupType operator()(
+            const detail::PolymorphicWrapper<interface::SetQuorum> &sq) const {
+          std::string class_name = "SetQuorum";
+          ReasonsGroupType res;
+
+          validateAccountId(res, sq->accountId());
+          validateQuorum(res, sq->newQuorum());
+
+          return res;
+        }
+
+        ReasonsGroupType operator()(
+            const detail::PolymorphicWrapper<interface::TransferAsset> &ta)
+            const {
+          std::string class_name = "SetQuorum";
+          ReasonsGroupType res;
+
+          validateAccountId(res, ta->srcAccountId());
+          validateAccountId(res, ta->destAccountId());
+          validateAssetId(res, ta->assetId());
+          validateAmount(res, ta->amount());
+
+          return res;
+        }
+
+       private:
+        void validateAccountId(
+            ReasonsGroupType &reason,
+            const interface::types::AccountIdType &account_id) const {
+          std::regex e("[a-z]{1,9}\\@[a-z]{1,9}");
+          if (not std::regex_match(account_id, e)) {
+            reason.second.push_back("Wrongly formed account_id");
+          }
+        }
+
+        void validateAssetId(
+            ReasonsGroupType &reason,
+            const interface::types::AssetIdType &asset_id) const {
+          std::regex e("[a-z]{1,9}\\#[a-z]{1,9}");
+          if (not std::regex_match(asset_id, e)) {
+            reason.second.push_back("Wrongly formed asset_id");
+          }
+        }
+
+        void validateAmount(ReasonsGroupType &reason,
+                            const interface::Amount &amount) const {
+          // what kind of validation could be here?
+        }
+
+        void validatePubkey(ReasonsGroupType &reason,
+                            const interface::types::PubkeyType &pubkey) const {
+          // what constraints can be here?
+        }
+
+        void validatePeerAddress(
+            ReasonsGroupType &reason,
+            const interface::AddPeer::AddressType &address) const {
+          // what constraints can be here?
```

Comment:
It's either ip or dns hostname, but for now I think it can be marked with todo (there's a dns check in develop afair)